### PR TITLE
job use future

### PR DESCRIPTION
**Pull request by panda-sheep**
fix comments

## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Description:
Use future to make meta jobs of different spaces execute in parallel
note:
Because the balance leader is for all spaces, it will not be dealt with here.
Same as https://github.com/vesoft-inc/nebula-ent/pull/1259
Also resolved issue https://github.com/vesoft-inc/nebula-ent/issues/867

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by SuperYoko — APPROVED** [review]

**Comment by kikimo** [comment]
cannot stop a failed job?

```txt
(root@nebula) [test]> show jobs;
+--------+------------------+------------+----------------------------+----------------------------+
| Job Id | Command          | Status     | Start Time                 | Stop Time                  |
+--------+------------------+------------+----------------------------+----------------------------+
| 14     | "ZONE_BALANCE"   | "FAILED"   | 2022-09-19T11:12:54.000000 | 2022-09-19T11:13:25.000000 |
| 13     | "ZONE_BALANCE"   | "FAILED"   | 2022-09-19T11:11:49.000000 | 2022-09-19T11:12:54.000000 |
| 12     | "LEADER_BALANCE" | "FINISHED" | 2022-09-19T09:29:16.000000 | 2022-09-19T09:29:21.000000 |
| 11     | "ZONE_BALANCE"   | "FINISHED" | 2022-09-19T09:29:21.000000 | 2022-09-19T11:11:49.000000 |
| 10     | "LEADER_BALANCE" | "FINISHED" | 2022-09-19T08:15:59.000000 | 2022-09-19T08:15:59.000000 |
| 9      | "ZONE_BALANCE"   | "FAILED"   | 2022-09-19T08:15:59.000000 | 2022-09-19T09:29:16.000000 |
| 8      | "LEADER_BALANCE" | "FINISHED" | 2022-09-19T08:14:18.000000 | 2022-09-19T08:14:23.000000 |
| 7      | "ZONE_BALANCE"   | "FINISHED" | 2022-09-19T08:14:59.000000 | 2022-09-19T08:15:59.000000 |
| 6      | "LEADER_BALANCE" | "FINISHED" | 2022-09-19T08:07:32.000000 | 2022-09-19T08:07:37.000000 |
| 5      | "LEADER_BALANCE" | "FINISHED" | 2022-09-19T08:05:26.000000 | 2022-09-19T08:05:31.000000 |
| 4      | "ZONE_BALANCE"   | "FINISHED" | 2022-09-19T08:05:31.000000 | 2022-09-19T08:07:32.000000 |
| 3      | "LEADER_BALANCE" | "FINISHED" | 2022-09-19T08:04:04.000000 | 2022-09-19T08:04:09.000000 |
| 2      | "ZONE_BALANCE"   | "STOPPED"  | 2022-09-19T08:07:37.000000 | 2022-09-19T08:10:26.000000 |
+--------+------------------+------------+----------------------------+----------------------------+
Got 13 rows (time spent 2745/3430 us)

Tue, 20 Sep 2022 10:11:12 CST

(root@nebula) [test]> stop job 14;
[ERROR (-1005)]: Finished job or failed job can not be stopped, please start another job instead

Tue, 20 Sep 2022 10:11:17 CST

```

**Comment by panda-sheep** [comment]
> cannot stop a failed job?

When I tested it in the enterprise version, it was OK.
Maybe a pr is not synced to the community version.

I take a look. thx

**Comment by panda-sheep** [comment]
https://github.com/vesoft-inc/nebula-ent/pull/861.  this pr is not synced to the community version. 
I will sync it

**Comment by liwenhui-soul** [comment]
generally LGTM

**Review by liwenhui-soul — DISMISSED** [review]

**Review by kikimo — DISMISSED** [review]
verified, LGTM

**Comment by panda-sheep** [comment]
Don't need merge first, wait for @critical27  to take a look

File: src/meta/test/BalancerTest.cpp [hunk 1]
```diff
@@ -912,21 +887,18 @@ TEST(BalanceTest, StopPlanTest) {
...
-  folly::Baton<true, std::atomic> baton;
-  balancer.setFinishCallBack([&](meta::cpp2::JobStatus) {
-    baton.post();
-    return nebula::cpp2::ErrorCode::SUCCEEDED;
-  });
-  auto ret = balancer.executeInternal();
-  EXPECT_EQ(nebula::cpp2::ErrorCode::SUCCEEDED, ret.value());
+  balancer.lostZones_ = {"5"};
+
+  auto ret = balancer.executeInternal().get();
```
Comment (critical27) [inline src/meta/test/BalancerTest.cpp:897]: Sorry for late response, would you try to don't call in sync here. We could call `ret = balancer.executeInternal();` and `balancer.stop()`. Then call `ret.wait()`.

This test is supposed to test stop the running tasks, if modify in this way, it is meaningless actually.

**Review by critical27 — COMMENTED** [review]
Good job~

File: src/meta/test/BalancerTest.cpp (hunk 1 above)
Comment (panda-sheep) [reply src/meta/test/BalancerTest.cpp]: Has fixed.

**Review by critical27 — APPROVED** [review]
Well done

**Review by kikimo — APPROVED** [review]
LGTM

**Pull request merged**